src/react/index.tsx

```diff
@@ -172,7 +172,7 @@ export async function getProviders() {
 export async function signIn<
   P extends RedirectableProvider | undefined = undefined
 >(
-  provider?: RedirectableProvider,
+  provider?: string,
```

Comment:
This seems to destroy the purpose of RedirectableProvider.

What we would like here is to hint the user of any of the built-in provider types, but not restricting it to those.